src/tools/tests/golden.arcs

```diff
@@ -16,3 +16,10 @@ particle Gold
   alias: writes {val: Text}
 
   collection: reads [{num: Number}]
+
+  typeVarIn: reads ~a with {amt: Number}
```

Comment:
FYI we should slowly be moving to better unit tests above the goldens. Our reliance on golden files is not healthy, as you cannot easily test many variations without having goldens generate multiple thousands of lines and become unbearable.

As you probably noted I've refactored schema2kotlin to allow unit testing of various pieces of the code generated particle base class. We should do the same for the code generated entity classes. Then we will be able to write unit tests specific to the cases we want to check.